### ISSUE TITLE
The process never exits although Got has resolved

### ISSUE THREAD

**Issue by CatsMiaow** [sindresorhus/got#1196]
- Node.js v12.16.1
- Got v11.0.2
- Jest v25.4.0

No problem with Got v10.7.0
It happens in v11

**test.spec.ts**
```ts
import got from 'got';

test('test', async () => {
  const data = await got('https://www.google.com');

  expect(typeof data).toBe('object');
  data.destroy(); // The same problem occurs without this code.
});
```
![image](https://user-images.githubusercontent.com/1300172/80353189-f3410380-88af-11ea-90e7-4c3c2011121d.png)

https://github.com/facebook/jest/issues/6423

**Comment by szmarczak** [comment]
Sorry, we don't use Jest. This looks similar to https://github.com/sindresorhus/got/pull/469/files#diff-5a1eff5599ddf2d36f1eb8ab13b82f82R1-R9

Since `process._getActiveHandles`, we need to use `async_hooks` instead.

**Comment by szmarczak** [comment]
oh lemme see the stacktrace first

**Comment by szmarczak** [comment]
https://github.com/szmarczak/cacheable-lookup/blob/d6d02f5ef66ed117244d8fbca8982320a90ff233/source/index.js#L97

**Comment by szmarczak** [comment]
Node.js v13.13.0
got@11.0.2 
cacheable-lookup@4.1.2
jest@25.4.0

```
szm@solus ~/Desktop/demo $ npx jest
 PASS  __tests__/x.js
  ✓ test (254ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        0.998s
Ran all test suites.
```

**Comment by szmarczak** [comment]
I think this is that:

```js
const got = require('got');

got('https://google.com', {dnsCache: false}).then(() => {
    console.log('should exit now');
});
```

The process never exits...

**Comment by szmarczak** [comment]
You can actually replace the whole thing with `new (require('cacheable-lookup'))`.

**Comment by szmarczak** [comment]
https://github.com/szmarczak/cacheable-lookup/issues/17

**Comment by szmarczak** [comment]
Fixed in `cacheable-lookup@4.2.1`.
